### PR TITLE
feat: enable custom apiVersion

### PR DESCRIPTION
**Pull request by schneesu**

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/bk-bcs?pullRequest=1693) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/bk-bcs?pullRequest=1693) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/bk-bcs?pullRequest=1693) it.</sub>

File: bcs-services/cluster-resources/pkg/resource/constants.go [hunk 1]
```diff
@@ -83,9 +83,6 @@ const (
 )
 
 const (
-	// DefaultCJGroupVersion 特殊指定的 CronJob 版本
-	DefaultCJGroupVersion = "batch/v1beta1"
-
 	// DefaultHPAGroupVersion 特殊指定的 HPA 版本
 	DefaultHPAGroupVersion = "autoscaling/v2beta2"
```
Comment (schneesu) [inline bcs-services/cluster-resources/pkg/resource/constants.go:87]: 再看看能不能把 HPA 默认 ApiVersion 去了

File: bcs-services/cluster-resources/pkg/action/resource/resource.go [hunk 2]
```diff
@@ -93,9 +95,11 @@ func (m *ResMgr) Update(
 	if err != nil {
 		return nil, err
 	}
-	if err := m.checkAccess(ctx, namespace, manifest); err != nil {
+	if err = m.checkAccess(ctx, namespace, manifest); err != nil {
 		return nil, err
 	}
+	// apiVersion 以 manifest 中的为准，不强制要求 preferred
+	m.GroupVersion = mapx.Get(manifest, "apiVersion", "").(string)
```
Comment (jamesgetx) [inline bcs-services/cluster-resources/pkg/action/resource/resource.go:102]: 看要不要在mapx里面定义几个常用的类型，比如GetStr, GetInt, GetBool之类的, 就不用再`.(string)`, 如果有必要，也可以安全断言下，防止直接panic

File: bcs-services/cluster-resources/pkg/handler/hpa/hpa.go [hunk 3]
```diff
@@ -57,7 +63,7 @@ func (h *Handler) GetHPA(
 func (h *Handler) CreateHPA(
 	ctx context.Context, req *clusterRes.ResCreateReq, resp *clusterRes.CommonResp,
 ) (err error) {
-	resp.Data, err = resAction.NewResMgr(req.ClusterID, res.DefaultHPAGroupVersion, res.HPA).Create(
+	resp.Data, err = resAction.NewResMgr(req.ClusterID, "", res.HPA).Create(
```
Comment (jamesgetx) [inline bcs-services/cluster-resources/pkg/handler/hpa/hpa.go:66]: Nit: 对于空串的阅读，总会很困惑(苦于golang中没有关键字参数和默认值)：一个是不知道它是什么字段，另外一个就是空串背后的逻辑

Comment (schneesu) [reply bcs-services/cluster-resources/pkg/handler/hpa/hpa.go:66]: image

IDE 的话会做一些提示的支持，Github 不知有无相关插件

File: bcs-services/cluster-resources/pkg/action/resource/resource.go (hunk 2 above)
Comment (schneesu) [reply bcs-services/cluster-resources/pkg/action/resource/resource.go:102]: 好建议，我单独 PR 全部扫一下

**Review by jamesgetx — APPROVED** [review]

**Pull request merged**